app/src/main/java/com/example/android/sunshine/app/DetailFragment.java

```diff
@@ -193,37 +193,46 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
             mFriendlyDateView.setText(friendlyDateText);
             mDateView.setText(dateText);
 
-            // Read description from cursor and update view
-            String description = data.getString(COL_WEATHER_DESC);
+            // Get description from weather condition ID
+            String description = Utility.getStringForWeatherCondition(getActivity(), weatherId);
             mDescriptionView.setText(description);
+            mDescriptionView.setContentDescription(getString(R.string.a11y_forecast, description));
 
-            // For accessibility, add a content description to the icon field
-            mIconView.setContentDescription(description);
+            // For accessibility, add a content description to the icon field. Because the ImageView
+            // is independently focusable, it's better to have a description of the image. Using
+            // null is appropriate when the image is purely decorative or when the image already
+            // has text describing it in the same UI component
+            mIconView.setContentDescription(getString(R.string.a11y_forecast_icon, description));
 
             // Read high temperature from cursor and update view
             boolean isMetric = Utility.isMetric(getActivity());
 
             double high = data.getDouble(COL_WEATHER_MAX_TEMP);
             String highString = Utility.formatTemperature(getActivity(), high);
             mHighTempView.setText(highString);
+            mHighTempView.setContentDescription(getString(R.string.a11y_high_temp, highString));
```

Comment:
You are correct. Sometimes it's desired to set different CD than the text -
for succinctness or to provide more context.

Generally I would recommend making all these not important for
accessibility and putting a content description on the whole card/row.

On Wed, 20 Apr 2016, 13:46 Paul Blundell, <notifications@github.com> wrote:

> In app/src/main/java/com/example/android/sunshine/app/DetailFragment.java
> <https://github.com/eduardb/Sunshine-Version-2/pull/1#discussion_r60399889>
> :
>
> >
> >              // Read high temperature from cursor and update view
> >              boolean isMetric = Utility.isMetric(getActivity());
> >
> >              double high = data.getDouble(COL_WEATHER_MAX_TEMP);
> >              String highString = Utility.formatTemperature(getActivity(), high);
> >              mHighTempView.setText(highString);
> > +            mHighTempView.setContentDescription(getString(R.string.a11y_high_temp, highString));
>
> @ataulm <https://github.com/ataulm> correct me if I am wrong, but if a
> view has setText then this text would be used for the content description
> any way?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/eduardb/Sunshine-Version-2/pull/1/files/eafffa1bb4c0d605c06206d992863949083fe22d#r60399889>
>